app/models/transaction.rb

```diff
@@ -51,15 +51,34 @@ class Transaction < ActiveRecord::Base
   after_destroy :recalculate_amount
   after_restore :recalculate_amount
 
-  def self.custom_dates
-    [
-      ["Current month: #{TimeRange.format(Time.now, 'current')}", "current_month"],
-      ["Previous month: #{TimeRange.format(Time.now, 'prev_month')}", "prev_month"],
-      ["Last 3 months: #{TimeRange.format(Time.now, 'last_3')}", "last_3_months"],
-      ["Quarter: #{TimeRange.format(Time.now, 'quarter')}", "quarter"],
-      ["This year: #{TimeRange.format(Time.now, 'year')}", "this_year"],
-      ["Custom", "custom"]
-    ]
+  class << self
+    def flow_ordered(def_currency)
+      currencies = Currency.ordered(def_currency)
+
+      curr_flow = {}
+      currencies.each do |curr|
+        curr_flow[curr] = {}
+
+        transactions = by_currency(curr)
```

Comment:
@Gataniel I think it will be better to make this via sql with one query. It will work much faster.